crates/ty_completion_eval/src/main.rs

```diff
@@ -0,0 +1,588 @@
+    let tmp_eval_dir = tmpdir.join("ty-completion-eval");
+    std::fs::create_dir_all(tmp_eval_dir.as_std_path())
+        .with_context(|| tmp_eval_dir.to_string())?;
+
+    let sources = TaskSource::all(&truth)?;
+    match args.command {
+        Command::ShowOne(ref cmd) => {
+            let Some(source) = sources
+                .iter()
+                .find(|source| cmd.matches_source_task(source))
+            else {
+                anyhow::bail!("could not find task named `{}`", cmd.task_name);
+            };
+            let tasks = source.to_tasks(&tmp_eval_dir)?;
+            let matching: Vec<&Task> = tasks.iter().filter(|task| cmd.matches_task(task)).collect();
+            anyhow::ensure!(
+                !matching.is_empty(),
+                "could not find any tasks matching the given criteria",
+            );
+            anyhow::ensure!(
+                matching.len() < 2,
+                "found more than one task matching the given criteria",
+            );
+            let task = &matching[0];
+            let completions = task.completions()?;
+
+            let mut stdout = std::io::stdout().lock();
+            for (i, c) in completions.iter().enumerate() {
+                write!(stdout, "{}", c.name.as_str())?;
+                if let Some(module_name) = c.module_name {
+                    write!(stdout, " (module: {module_name})")?;
+                }
+                if task.cursor.answer.matches(c) {
+                    write!(stdout, " (*, {}/{})", i + 1, completions.len())?;
+                }
+                writeln!(stdout)?;
+            }
+            writeln!(stdout, "-----")?;
+            writeln!(stdout, "found {} completions", completions.len())?;
+            Ok(ExitCode::SUCCESS)
+        }
+        Command::All(AllCommand { threshold, tasks }) => {
+            let mut precision_sum = 0.0;
+            let mut results_wtr = None;
+            if let Some(ref tasks) = tasks {
+                let mut wtr = csv::Writer::from_path(SystemPath::new(tasks))?;
+                wtr.serialize(("name", "file", "index", "rank"))?;
+                results_wtr = Some(wtr);
+            }
+            for source in &sources {
+                for task in source.to_tasks(&tmp_eval_dir)? {
+                    let completions = task.completions()?;
+                    let rank = task.rank(&completions)?;
+                    precision_sum += rank.map(|rank| 1.0 / f64::from(rank)).unwrap_or(0.0);
+                    if let Some(ref mut wtr) = results_wtr {
+                        wtr.serialize((
+                            &task.name,
+                            &task.cursor.path.as_str(),
+                            task.cursor.index,
+                            rank,
+                        ))?;
+                    }
+                }
+            }
+            #[allow(clippy::cast_precision_loss)]
+            let mrr = precision_sum / (sources.len() as f64);
```

Comment:
This seems wrong to me? Shouldn't we be normalizing by the total number of tasks, instead of the number of source files? If I split a file with multiple tasks across several files, the MRR should stay the same?